tests/test_config.py

```
@@ -791,5 +799,168 @@ def test_no_env_vars_returns_empty_dict(self):
+
+    @patch("os.path.exists", return_value=True)
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("json.load")
+    def test_load_credentials_success(self, mock_json_load, mock_open, mock_exists):
+        """Test successful credential loading from JSON file."""
+        mock_json_load.return_value = {"user_id": "test", "access_token": "token"}
+        result = load_credentials()
+        self.assertEqual(result, {"user_id": "test", "access_token": "token"})
+
+    @patch("os.makedirs", side_effect=OSError("Permission denied"))
+    def test_save_credentials_directory_creation_failure(self, mock_makedirs):
```

Ruff (0.13.3)

Unused method argument: `mock_makedirs`

(ARG002)